pkg/store/proxy.go

```diff
@@ -302,10 +304,26 @@ func (s *ProxyStore) Series(originalRequest *storepb.SeriesRequest, srv storepb.
 		WithoutReplicaLabels:    originalRequest.WithoutReplicaLabels,
 	}
 
+	// We may arrive here either via the promql engine
+	// or as a result of a grpc call in layered queries
+	var tenant string
+	ctx := srv.Context()
+	if ctx.Value(tenancy.TenantKey) != nil {
+		tenant = ctx.Value(tenancy.TenantKey).(string)
+	} else {
+		tenant, err = tenancy.GetTenantFromGRPCMetadata(ctx)
+		if err != nil {
+			level.Warn(s.logger).Log("msg", err)
+		}
+	}
```

Comment:
Let's make this a bit simpler. Also probably debug log is fine here, as request would get assigned default tenant anyway. 🙂 

Ditto for the rest.
```suggestion
	ctx := srv.Context()
	tenant, err := tenancy.GetTenantFromGRPCMetadata(ctx)
	if err != nil {
		level.Debug(s.logger).Log("msg", "using tenant from context instead of metadata")
		if ctx.Value(tenancy.TenantKey) != nil {
			tenant = ctx.Value(tenancy.TenantKey).(string)
		}
	}
```

Reply:
Fixed :+1: